### PR TITLE
#158845462 Seed datasets for database

### PR DESCRIPTION
**Pull request by Stylll**
#### What does this PR do?
Create migration feature for seeding data into the database. 

#### Description of Task to be completed?
- Add seed data files for questions and answers
- Create config file and migration files to facilitate migration
- Add npm scripts to package json

#### How should this be manually tested?
- run `npm install`
- update env file using .env.example file
- for migration to local db `npm run migrate`
- for migration to local test db `npm run migrate:test`
- for migration to production db `npm run migrate:prod`
- for migration to staging db `npm run migrate:staging`

- suffix script with '-down' to revert migration. e.g: `npm run migrate:down`

#### What are the relevant pivotal tracker stories?
#158845462

#### Questions:

### CODE REVIEW

**Review by nouwatinjacob — APPROVED** [review]
Good work. I love how you made your code to be readable by adding comments (where necessary) to indicate what it does. Keep it up!

File: package.json [hunk 1]
```diff
@@ -23,7 +23,15 @@
     "merge:lcov": "cat ./coverage/*/lcov.info > ./coverage/lcov.info",
     "coverage": "npm run merge:lcov && cat ./coverage/lcov.info | coveralls",
     "heroku-postbuild": "npm run build",
-    "watch:scss": "gulp watchSCSS"
+    "watch:scss": "gulp watchSCSS",
```
Comment (otseobande) [inline package.json]: Do we still use this task?  I thought we have moved watching scss file changes to the `gulp startDev` task.

File: questions-data.js [hunk 2]
```diff
@@ -0,0 +1,1303 @@
...
+    id: 260,
+    numAnswers: 2,
+    official: true,
+    text: "Listen, son. If you want to get involved with _, I won't stop you. Just steer clear of _.",
+    region: 'America'
+  }
+];
+
+module.exports = questions;
+
```
Comment (otseobande) [inline questions-data.js]: Do you think moving this file and the "answers-data.js" file to a folder called say `seed_data` be a better way to organise our seed data files?

File: config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  dbName = 'cfh';
+  dbUrl = process.env.MONGOHQ_URL;
+} else if (process.env.NODE_ENV === 'staging') {
+  dbName = 'gondor-cfh-staging';
+  dbUrl = process.env.MONGOHQ_URL_STAGING;
+} else {
+  dbName = 'gondor-cfh';
+  dbUrl = process.env.MONGOHQ_URL_PROD;
+}
+
```
Comment (otseobande) [inline config.js:18]: Isn't the `dbName` also conveyed in the `dbUrl`'s value? For example, assuming the`dbUrl` is `mongodb://localhost:27017/gondor_cfh` the last part of this url has the `dbName` included. Is there any particular reason why we need to specify the `dbName`?

File: config.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+module.exports = {
+
+  mongodb: {
+
+    // TODO Change (or review) the url to your MongoDB:
+    url: dbUrl,
+
+    // TODO Change this to your database name:
```
Comment (otseobande) [inline config.js]: Is there any reason we left the TODO comments?

File: package.json [hunk 5]
```diff
@@ -23,7 +23,15 @@
...
+    "migrate:db": "cross-env NODE_ENV=development ./node_modules/.bin/migrate-mongo up",
+    "migrate:db-test": "cross-env NODE_ENV=test ./node_modules/.bin/migrate-mongo up",
+    "migrate:db-staging": "cross-env NODE_ENV=staging ./node_modules/.bin/migrate-mongo up",
+    "migrate:db-prod": "cross-env NODE_ENV=production ./node_modules/.bin/migrate-mongo up",
+    "migrate:db-down": "cross-env NODE_ENV=development ./node_modules/.bin/migrate-mongo down",
+    "migrate:db-test-down": "cross-env NODE_ENV=test ./node_modules/.bin/migrate-mongo down",
+    "migrate:db-staging-down": "cross-env NODE_ENV=staging ./node_modules/.bin/migrate-mongo down",
+    "migrate:db-prod-down": "cross-env NODE_ENV=production ./node_modules/.bin/migrate-mongo down"
```
Comment (otseobande) [inline package.json]: I feel adding `db` before the migrate scripts makes them abit too long. I think `migrate:db` can be `migrate` and something like `migrate:db-test` can be `migrate:test`. Please let me know how you feel about it.

**Review by otseobande — COMMENTED** [review]
Nicely done. Using `migrate-mongo` is a great idea.  However, there are a few review comments I left behind. Please check them out.

File: package.json (hunk 1 above)
Comment (Stylll) [reply package.json]: Yes we have. I forgot to remove it. I'll remove that immediately. Thanks.

File: config.js (hunk 3 above)
Comment (Stylll) [reply config.js:18]: Yes there's a reason. The npm module I used to facilitate migration "migrate-mongo" requires that the dbName value be supplied even if the url already contains the database name. I've tried finding a way to work around it, all to no avail.

File: config.js (hunk 4 above)
Comment (Stylll) [reply config.js]: I left it there so it'll be easy for others to know how to make changes if necessary. However, looking at it again, the names are quite self explanatory so I'll take those comments out. Thanks for pointing that out.

File: package.json (hunk 5 above)
Comment (Stylll) [reply package.json]: I like that suggestion. It makes it easy to type and remember. I'll ensure to make this changes. Thanks.

File: questions-data.js (hunk 2 above)
Comment (Stylll) [reply questions-data.js]: Good idea!. It helps improve file organisation especially if the need arises for us to add more seed data files. I'll move the files into the suggested folder. Thanks.

**Review by otseobande — APPROVED** [review]

**Pull request merged**